src/cursor/cur_hs.c

```diff
@@ -511,133 +606,99 @@ __curhs_search_near(WT_CURSOR *cursor, int *exactp)
     WT_ERR(__wt_hs_cursor_search_near(session, file_cursor, &exact));
 
     /*
-     * There are some key fields missing so we are searching a range of keys. Place the cursor at
-     * the start of the range.
+     * If we have some missing fields in the key, try to walk forward first to increase the chance
```

Comment:
The comment is obsolete. I forgot to remove it. We should rephrase or remove it.